src/adapter/search-response-adapter/hits-adapter.ts

```diff
@@ -1,24 +1,30 @@
-import type { PaginationContext, SearchContext } from '../../types'
-import { adaptPagination } from './pagination-adapter'
+import type { SearchContext, MeiliSearchResponse } from '../../types'
 import { adaptFormattedFields } from './format-adapter'
 import { adaptGeoResponse } from './geo-reponse-adapter'
 
 /**
- * @param  {Array<Record<string} hits
+ * @param  {MeiliSearchResponse<Record<string, any>>} searchResponse
  * @param  {SearchContext} searchContext
- * @param  {PaginationContext} paginationContext
- * @returns {any}
+ * @returns {Array<Record<string, any>>}
  */
 export function adaptHits(
-  hits: Array<Record<string, any>>,
-  searchContext: SearchContext,
-  paginationContext: PaginationContext
+  searchResponse: MeiliSearchResponse<Record<string, any>>,
+  searchContext: SearchContext
 ): any {
   const { primaryKey } = searchContext
-  const { hitsPerPage, page } = paginationContext
-  const paginatedHits = adaptPagination(hits, page, hitsPerPage)
+  const { hits } = searchResponse
+  const {
+    pagination: { finite, page, hitsPerPage },
+  } = searchContext
 
-  let adaptedHits = paginatedHits.map((hit: Record<string, any>) => {
+  if (!finite) {
```

Comment:
Can you help me understand this piece of code, I'm not sure why you need it 👼

Reply:
```
 if (!finite) {
    // In case we are not in a finite environment
    // Hits contains all the documents of every page before the current one.
    const deleteCount = page * hitsPerPage
    // Removes the hits that are not included in the current pagination
    // Example, if we are on page 2, we need to remove the hits from page 1
    hits.splice(0, deleteCount)
    if (hits.length > hitsPerPage) {
      // Since we append the limit of one to know if there is another page
      // We need to remove the last hit in case the length of hits is bigger than the required hitsPerPage
      // Example: limit: 7, hitsPerPage: 6, Hits length: 7. We need to remove the last hit
      hits.splice(hits.length - 1, 1)
    }
  }

```

Is this better?